lib/vector-core/src/config/proxy.rs

```diff
@@ -204,11 +209,44 @@ mod tests {
         header::{AUTHORIZATION, PROXY_AUTHORIZATION},
         HeaderName, HeaderValue, Uri,
     };
+    use proptest::prelude::*;
 
     const PROXY_HEADERS: [HeaderName; 2] = [AUTHORIZATION, PROXY_AUTHORIZATION];
 
     use super::*;
 
+    impl Arbitrary for ProxyConfig {
```

Comment:
Nice 😄

Reply:
Why test the one field you know has an issue when you can test for all of them. I'm thinking to submit contributions to the other two types to support proptesting.